modules/core/src/main/scala/sangria/schema/Schema.scala

```diff
@@ -1529,10 +1529,33 @@ case class Schema[Ctx, Val](
 
     directImplementations.map { case (name, directImpls) =>
       name -> directImpls.flatMap(findDescendants).groupBy(_.name).map(_._2.head).toVector
-    }
+    } ++ unionTypes.values.map(ut => ut.name -> ut.types.toVector)
```

Comment:
Actually, I don't like this

Reply:
Do you have a better idea?

Reply:
Yeah I have changed back and use union only in possibleTypes